taskfile/read/taskfile.go

```diff
@@ -109,6 +110,12 @@ func Taskfile(dir string, entrypoint string) (*taskfile.Taskfile, error) {
 			}
 
 			for _, task := range includedTaskfile.Tasks {
+				if includedTask.Vars != nil {
+					if task.Vars == nil {
+						task.Vars = &taskfile.Vars{}
+					}
+					task.Vars.Merge(includedTask.Vars)
```

Comment:
I think the parent task file vars should take precedence over the included vars. Also, it might be useful to include the parent vars in the included.

```go
included.Vars.Merge(task.Vars)
task.Vars.Merge(included.Vars)
```

This behavior should be extracted to a separate function and have a couple tests written for it as well. This will also document what we expect to happen, and allow us to write some scenarios to see if we can find edge cases that might behave strangely.

Reply:
I was thinking that the expected behavior should be that if there are vars in the included taskfile then they should be used unless overridden by the vars specified to the include.  I don't think that root taskfile vars should be applied because if you wanted them to override the included files vars you would put them in the vars map on the include statement.
I will go ahead and work up some tests. I wanted to get this posted to see if there is any opposition in the idea of doing this.

Reply:
I added a test to the root tests to show how it works and how I expect it to work.

Reply:
I can help with other tests if you let me know the requirements that you are looking for and where you want them to go

Reply:
Oh, you know, I think I missed that this _adds_ vars to the `include`. All we need is for the included file to have access to the parent vars, right?

Reply:
If you look at the test I added it shows off the functionality that I'm after. I want to set the same variable to different values for each include

Reply:
@ghostsquad  Is there anything else that is needed to get this considered?  I would like to start using this if possible.

Reply:
> Oh, you know, I think I missed that this _adds_ vars to the `include`. All we need is for the included file to have access to the parent vars, right?

I think the way the PR implements it is much more useful. This way you can properly re-use tasks with a different set of variables.

It would be great to have this!

Reply:
The way `includes` currently reads, because that it's a `namespace` (tasks in an includes are automatically prefixed), the `included` Taskfile is "more specific" than the parent task file. To quote the documentation:

> When doing interpolation of variables, Task will look for the below. They are listed below in order of importance (e.g. most important first):

    1. Variables declared in the task definition
    2. Variables given while calling a task from another (See [Calling another task](https://taskfile.dev/#/usage?id=calling-another-task) above)
    3. Global variables (those declared in the vars: option in the Taskfile)
    4. Environment variables
    
An exercise in appending this would be:

```diff
    1. Variables declared in the task definition
    2. Variables given while calling a task from another (See [Calling another task](https://taskfile.dev/#/usage?id=calling-another-task) above)
+   3. Global Vars within the _included_ Taskfiles
    4. Global variables (those declared in the vars: option in the Taskfile)
    5. Environment variables
```

The more I think about this, the current mental model of Task is such of passing vars from high to low. The highest point is the parent Taskfile.

If I'm reading `merge` correctly, then it really is an override.